### PR TITLE
fix(AdvancedSearchMenu): allow spaces in search form field values (#8369)

### LINKED ISSUE #8369: Bug - [SearchInput] - [When typing in advanced search input fields space it gets eaten]
**Describe the problem**
Advanced search input will eat spaces.

**How do you reproduce the problem?**
https://www.patternfly.org/v4/components/search-input/#advanced)

Try typing a string with a space in one of the input form fields. The space get's eaten.

**Expected behavior**
Spaces are needed in some cases. In ccokpit  we have a scenario where we need to provide daytime, which includes a space delimiter.
Ex: "2022-11-04 12:34:56"

**Is this issue blocking you?**
There is not any current workaround

### PR DESCRIPTION
**Pull request by KKoukiou**
When the user uses a space in a form field value this will get wrapped in quotes within the Search input.

Closes #8369

### CODE REVIEW

**Comment by KKoukiou** [comment]
Once I get some ack for the actual change I will deal with the tests.

File: packages/react-core/src/components/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -19,7 +19,8 @@ import { Popper } from '../../helpers';
 
 export interface SearchAttribute {
   /** The search attribute's value to be provided in the search input's query string.
-   * It should have no spaces and be unique for every attribute.
+   * It should be unique for every attribute. If spaces are used they will get wrapped
+   * in single quotes in the search input string.
```
Comment (nicolethoen) [inline packages/react-core/src/components/SearchInput/SearchInput.tsx:23]: I don't think this is the correct prop description to update. This is referring to the attribute names which should be unique with no spaces. Like 'username', not the values associated with usernames.

File: packages/react-core/src/components/SearchInput/SearchInput.tsx [hunk 2]
```diff
@@ -229,13 +230,28 @@ const SearchInputBase: React.FunctionComponent<SearchInputProps> = ({
     setIsSearchMenuOpen(false);
   };
 
+  const splitStringExceptInQuotes = (str: string) =>
+    str.match(/\\?.|^$/g).reduce(
+      (p: any, c: string) => {
+        if (c === "'") {
```
Comment (nicolethoen) [inline packages/react-core/src/components/SearchInput/SearchInput.tsx:236]: Can this be made to work with double quotes, too?

File: packages/react-core/src/components/SearchInput/SearchInput.tsx (hunk 1 above)
Comment (KKoukiou) [reply packages/react-core/src/components/SearchInput/SearchInput.tsx:23]: The attribute names are just bellow, the 'displayName' property. This is the value of the attributes in the form field.

So I document that if someone uses spaces in the attribute values, these will get wrapped in quotes within the search string.

Open to change it if you have some suggestion!

Comment (nicolethoen) [reply packages/react-core/src/components/SearchInput/SearchInput.tsx:23]: actually, the attribute names and the display names are all part of the same concept related to attributes, not the attribute values. 

This way, when someone wants a field on the advanced search form to be displayed as `User name` (with a space) but then the corresponding attribute displayed in the input before the value is using the attr prop to display `username`. so neither the SearchAttribute's `attr` field, nor its `display` field have anything to do with the associated values.

Comment (nicolethoen) [reply packages/react-core/src/components/SearchInput/SearchInput.tsx:23]: I don't think we have a place where the docs state any rules about the values themselves having no spaces

**Review by mcarrano — COMMENTED** [review]
I think this makes sense. My only question is whether single or double quotes are more appropriate. @KKoukiou I see you use single. Was there a reason for this? I also feel like this calls for some documentation to let consumers know that this feature exists.

**Comment by nicolethoen** [comment]
Maybe a sentence or two could be added above the example in the markdown explaining the intended usage of quotes?

**Comment by KKoukiou** [comment]
> I think this makes sense. My only question is whether single or double quotes are more appropriate. @KKoukiou I see you use single. Was there a reason for this? I also feel like this calls for some documentation to let consumers know that this feature exists.

Both single and double quotes are accepted actually :) cockpit has a usecase where space is required in the advanced search form fields.

**Review by nicolethoen — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**